diff --git a/libbb/alloc_affinity.c b/libbb/alloc_affinity.c
index b6d9f64..3567ab9 100644
--- a/libbb/alloc_affinity.c
+++ b/libbb/alloc_affinity.c
@@ -14,6 +14,15 @@ unsigned long* FAST_FUNC get_malloc_cpu_affinity(int pid, unsigned *sz)
 	unsigned long *mask = NULL;
 	unsigned sz_in_bytes = *sz;
 
+#if defined(__seele__)
+	(void) pid;
+	if (sz_in_bytes < sizeof(unsigned long))
+		sz_in_bytes = sizeof(unsigned long);
+	mask = xzalloc(sz_in_bytes);
+	mask[0] = 1;
+	*sz = sz_in_bytes;
+	return mask;
+#else
 	for (;;) {
 		mask = xrealloc(mask, sz_in_bytes);
 		if (sched_getaffinity(pid, sz_in_bytes, (void*)mask) == 0)
@@ -26,4 +35,5 @@ unsigned long* FAST_FUNC get_malloc_cpu_affinity(int pid, unsigned *sz)
 	//bb_error_msg("get mask[0]:%lx sz_in_bytes:%d", mask[0], sz_in_bytes);
 	*sz = sz_in_bytes;
 	return mask;
+#endif
 }
