diff --git a/libbb/xfuncs_printf.c b/libbb/xfuncs_printf.c
index 842d10c..9c53bb3 100644
--- a/libbb/xfuncs_printf.c
+++ b/libbb/xfuncs_printf.c
@@ -726,8 +726,14 @@ int FAST_FUNC wait_for_exitstatus(pid_t pid)
 
 void FAST_FUNC xsettimeofday(const struct timeval *tv)
 {
+#if defined(__seele__)
+	(void) tv;
+	errno = ENOSYS;
+	bb_simple_perror_msg_and_die("settimeofday");
+#else
 	if (settimeofday(tv, NULL))
 		bb_simple_perror_msg_and_die("settimeofday");
+#endif
 }
 
 void FAST_FUNC xgettimeofday(struct timeval *tv)
